lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -141,6 +141,16 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # therefore, this command will always exit 0 if either service is installed
             on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
 
+            # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
+            # registry and honored after install/upgrade.
+            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
```

Comment:
I don't think that this is quite right since we also need to look in a different location on a 32-bit OS

```
HKLM\Software\Puppet Labs\PuppetInstaller
```

https://github.com/puppetlabs/puppet_for_the_win/blob/ae1821b2ff7ab6c417bfb03b89ab3097be3e05fa/README.markdown#remembered-properties